### PR TITLE
Tweaks to permit non-Yotta build

### PR DESCRIPTION
**Pull request by kjbracey-arm**
Various start-up changes to permit the same basic interface bring-up
to also work outside the Yotta system.

### CODE REVIEW

**Comment by kjbracey-arm** [comment]
@artokin, @SeppoTakalo, @c1728p9

**Comment by SeppoTakalo** [comment]
+1

File: source/mesh_system.c [hunk 1]
```diff
@@ -61,10 +65,15 @@ void mesh_system_init(void)
         randLIB_seed_random();
         platform_timer_enable();
         eventOS_scheduler_init();
+#ifndef YOTTA_CFG
+        ns_event_loop_thread_create();
+#endif
         trace_init(); // trace system needs to be initialized right after eventOS_scheduler_init
```
Comment (c1728p9) [inline source/mesh_system.c]: does trace_init() need to be moved to before ns_event_loop_thread_create()?

Comment (kjbracey-arm) [inline source/mesh_system.c]: Don't think it matters (at least at the moment) - that only creates the thread, it doesn't start it. One-line function, without trace.

Guess I was viewing that thread_create call as an extension of the eventOS_scheduler_init().

Comment (kjbracey-arm) [inline source/mesh_system.c]: (If there was a platform event_init function that eventOS_scheduler_init called, it could maybe have been put there)

**Comment by c1728p9** [comment]
Looks good to me

**Pull request merged**